frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/RebaseOnParentBackgroundable.java

```diff
@@ -0,0 +1,174 @@
+package com.virtuslab.gitmachete.frontend.actions.backgroundables;
+
+import static com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle.getString;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.concurrent.atomic.AtomicReference;
+
+import com.intellij.openapi.progress.ProgressIndicator;
+import com.intellij.openapi.progress.Task;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vcs.VcsNotifier;
+import git4idea.branch.GitRebaseParams;
+import git4idea.config.GitVersion;
+import git4idea.rebase.GitRebaseOption;
+import git4idea.rebase.GitRebaseUtils;
+import git4idea.repo.GitRepository;
+import git4idea.util.GitFreezingProcess;
+import io.vavr.control.Option;
+import io.vavr.control.Try;
+import kr.pe.kwonnam.slf4jlambda.LambdaLogger;
+import lombok.CustomLog;
+import lombok.experimental.ExtensionMethod;
+import lombok.val;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import com.virtuslab.gitmachete.backend.api.IGitMacheteRepositorySnapshot;
+import com.virtuslab.gitmachete.backend.api.IGitRebaseParameters;
+import com.virtuslab.gitmachete.backend.api.INonRootManagedBranchSnapshot;
+import com.virtuslab.gitmachete.backend.api.hooks.IExecutionResult;
+import com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle;
+import com.virtuslab.qual.guieffect.UIThreadUnsafe;
+
+@ExtensionMethod(GitMacheteBundle.class)
+@CustomLog
+public class RebaseOnParentBackgroundable extends Task.Backgroundable {
+
+  private final GitRepository gitRepository;
+
+  private static final String NL = System.lineSeparator();
+  private final IGitMacheteRepositorySnapshot gitMacheteRepositorySnapshot;
+
+  private final INonRootManagedBranchSnapshot branchToRebase;
+
+  private final Try<IGitRebaseParameters> tryGitRebaseParameters;
+
+  private final boolean shouldExplicitlyCheckout;
+
+  private final Project project;
+
+  public RebaseOnParentBackgroundable(Project project, String title, GitRepository gitRepository,
+      IGitMacheteRepositorySnapshot gitMacheteRepositorySnapshot,
+      INonRootManagedBranchSnapshot branchToRebase,
+      boolean shouldExplicitlyCheckout) {
+    super(project, title);
+    this.project = project;
+    this.gitRepository = gitRepository;
+    this.branchToRebase = branchToRebase;
+    this.gitMacheteRepositorySnapshot = gitMacheteRepositorySnapshot;
+    this.shouldExplicitlyCheckout = shouldExplicitlyCheckout;
+    LOG.debug(() -> "Entering: project = ${project}, gitRepository = ${gitRepository}, branchToRebase = ${branchToRebase}");
+
+    tryGitRebaseParameters = Try.of(branchToRebase::getParametersForRebaseOntoParent);
```

Comment:
Nit: maybe just pass ready `gitRebaseParameters` to the ctor? 🤔

Reply:
I am leaving the try call in this class as it provides nice handling for the try however, I managed to improve the code a bit